test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/PackageItemLoaderTests.cs

```diff
@@ -25,222 +22,82 @@
-
-            var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(new[] { source1, source2 });
-            var repositories = sourceRepositoryProvider.GetRepositories();
-
-            var context = new PackageLoadContext(repositories, false, uiContext);
-
-            var packageFeed = new MultiSourcePackageFeed(repositories, logger: null, telemetryService: null);
-            var loader = new PackageItemLoader(context, packageFeed, "nuget");
+                PackageSearchMetadataContextInfo.Create(packageSearchMetadata)
+            };
 
-            var totalCount = await loader.GetTotalCountAsync(100, CancellationToken.None);
+            var searchResult = new SearchResultContextInfo(packageSearchMetadataContextInfo, new Dictionary<string, LoadingStatus> { { "Completed", LoadingStatus.Ready } }, false);
 
-            Assert.NotInRange(totalCount, 0, 99);
-        }
+            searchService.Setup(x =>
+                x.SearchAsync(
+                    It.IsAny<IReadOnlyCollection<IProjectContextInfo>>(),
+                    It.IsAny<IReadOnlyCollection<PackageSourceContextInfo>>(),
+                    It.IsAny<string>(),
+                    It.IsAny<SearchFilter>(),
+                    It.IsAny<NuGet.VisualStudio.Internal.Contracts.ItemFilter>(),
+                    It.IsAny<bool>(),
+                    It.IsAny<CancellationToken>()))
+                .Returns(new ValueTask<SearchResultContextInfo>(searchResult));
 
-        [Fact]
-        public async Task LoadNextAsync_Works()
-        {
-            var solutionManager = Mock.Of<INuGetSolutionManagerService>();
-            var uiContext = new Mock<INuGetUIContext>();
+            searchService.Setup(x =>
+                x.RefreshSearchAsync(It.IsAny<CancellationToken>()))
+                .Returns(new ValueTask<SearchResultContextInfo>(searchResult));
 
             uiContext.Setup(x => x.SolutionManagerService)
                 .Returns(solutionManager);
 
             uiContext.Setup(x => x.ServiceBroker)
                 .Returns(Mock.Of<IServiceBroker>());
 
-            var context = new PackageLoadContext(sourceRepositories: null, isSolution: false, uiContext.Object);
-            var packageFeed = new TestPackageFeed();
-            var loader = new PackageItemLoader(context, packageFeed, TestSearchTerm, includePrerelease: true);
-
-            Assert.Equal(LoadingStatus.Unknown, loader.State.LoadingStatus);
-            var initial = loader.GetCurrent();
-            Assert.Empty(initial);
-
-            var loaded = new List<PackageItemListViewModel>();
-
-            await loader.LoadNextAsync(progress: null, CancellationToken.None);
+            var source1 = new PackageSourceContextInfo("https://pkgs.dev.azure.com/dnceng/public/_packaging/nuget-build/nuget/v3/index.json", "NuGetBuild");
+            var source2 = new PackageSourceContextInfo("https://api.nuget.org/v3/index.json", "NuGet.org");
```

Comment:
Can we not use real package sources?  We've had tests work for years, then start failing when the assumptions about what packages are/aren't in the source change, blocking the dev branch. Our unit tests in particular should not be integration tests with external implementations of the nuget protocol.

Reply:
tracking with https://github.com/NuGet/Home/issues/10393